pilot/pkg/model/xds_cache.go

```diff
@@ -164,10 +164,14 @@ type lruCache struct {
 	store            simplelru.LRUCache
 	// token stores the latest token of the store, used to prevent stale data overwrite.
 	// It is refreshed when Clear or ClearAll are called
-	token       CacheToken
-	mu          sync.RWMutex
-	configIndex map[ConfigKey]sets.Set
-	typesIndex  map[config.GroupVersionKind]sets.Set
+	token CacheToken
+	mu    sync.RWMutex
+	// configIndex and typesIndex stores the xds key pointer,
+	// because different configs or different kinds of resources can be part of a same xds key.
+	// Say route `80` can be influenced by hundreds of services with port 80.
```

Comment:
Since we take md5 hash, its always constant size?

Reply:
the string should be constant size, yes, because we use take the md5 hash.

what @hzxuzhonghu was trying to do was save a half the memory by using a pointer, a string variable uses 16bytes (pointer to byte array and int for the length), and a pointer to a string is only 8 bytes.

the set stores the keys, so using a pointer as key would mean that those keys use half the amount of memory.

What we don't know is how much is 8bytes per string saving compared to the amount of bytes the rest of the cache is using.
Also the string objects still remain in memory because we reference them, so in some cases we may actually be wasting more memory? (pointer+string in memory). Though we may add the same string to multiple sets, so in that case, string pointers could save some memory.

Reply:
We may save a bit but I am not really convinced its worth this, and also agree same concern that we still have to store the string. I think the underlying issue in the main issue is pretty clearly a leak on the configindex - micro-optimizations like this would not solve this case and don't seem to help much in general based on the benchmarks we do have. If there is some data that shows otherwise that would help

Reply:
It is worth, you can add hundreds of services or other dependencies in your benchmark test to see the diff.

Reply:
the test does that afaik. Feel free to tweak the benchmark though

Reply:
Nope， i can add that